lib/mocha/any_instance_method.rb

```diff
@@ -23,9 +23,9 @@ def restore_original_method
     end
 
     def method_visibility
-      (original_method_owner.public_instance_methods(true).include?(method_name) && :public) ||
-        (original_method_owner.protected_instance_methods(true).include?(method_name) && :protected) ||
-        (original_method_owner.private_instance_methods(true).include?(method_name) && :private)
+      (original_method_owner.public_method_defined?(method_name) && :public) ||
+        (original_method_owner.protected_method_defined?(method_name) && :protected) ||
+        (original_method_owner.private_method_defined?(method_name) && :private)
```

Comment:
> I used Travis to test this change against all the versions of Ruby we support and nothing breaks so I'm fairly confident that the old and new implementations are identical.

Hmm. I vaguely remember there being some difference in these possibly for different Ruby versions...